x-pack/plugins/actions/server/authorization/actions_authorization.ts

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import Boom from 'boom';
+import { KibanaRequest } from 'src/core/server';
+import { SecurityPluginSetup } from '../../../security/server';
+import { ActionsAuthorizationAuditLogger } from './audit_logger';
+import { ACTION_SAVED_OBJECT_TYPE, ACTION_TASK_PARAMS_SAVED_OBJECT_TYPE } from '../saved_objects';
+
+export interface ConstructorOptions {
+  request: KibanaRequest;
+  auditLogger: ActionsAuthorizationAuditLogger;
+  authorization?: SecurityPluginSetup['authz'];
+}
+
+const operationAlias: Record<
+  string,
+  (authorization: SecurityPluginSetup['authz']) => string | string[]
+> = {
+  execute: (authorization) => [
+    authorization.actions.savedObject.get(ACTION_SAVED_OBJECT_TYPE, 'get'),
+    authorization.actions.savedObject.get(ACTION_TASK_PARAMS_SAVED_OBJECT_TYPE, 'create'),
+  ],
+  list: (authorization) => authorization.actions.savedObject.get(ACTION_SAVED_OBJECT_TYPE, 'get'),
+};
+
+export class ActionsAuthorization {
+  private readonly request: KibanaRequest;
+  private readonly authorization?: SecurityPluginSetup['authz'];
+  private readonly auditLogger: ActionsAuthorizationAuditLogger;
+
+  constructor({ request, authorization, auditLogger }: ConstructorOptions) {
+    this.request = request;
+    this.authorization = authorization;
+    this.auditLogger = auditLogger;
+  }
+
+  public async ensureAuthorized(operation: string, actionTypeId?: string) {
+    const { authorization } = this;
+    if (authorization) {
```

Comment:
I didn't realise that was a valid state.
Is there a way for Kibana to *know* that's the case?

Reply:
You can ask the security plugin if Security is enabled in ES:

```ts
public setup(core: CoreSetup, plugins) {
  const { license } = plugins.security;
  const isSecurityEnabledInEs = license.isEnabled();
}
```

This is a point-in-time check, and won't give you an honest answer during `setup` because our connection to ES hasn't been established yet, but you can safely use this during the request lifecycle, as everything will be up and running by then. Sorry I didn't consider this case earlier, either

Reply:
🙏

Reply:
cool, this is now addressed :)